### ISSUE TITLE
Redshift dialect does not support scoped GRANT statements

### ISSUE THREAD

**Issue by adam-eljasiak** [sqlfluff/sqlfluff#8401]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff can't parse Redshift SQL scripts containing scoped GRANT statements.

### Expected Behaviour

Redshift allows to grant some/all privileges for all objects on the DATABASE or SCHEMA level, as described in https://docs.aws.amazon.com/redshift/latest/dg/r_GRANT.html, under `Granting scoped permissions` section. I'd expect SQLFluff to be able to parse these statements, so scripts containing them can enjoy all the goodness `sqlfluff fix` provides.

### Observed Behaviour

Running `sqlfluff fix example.sql` on a script with scoped GRANT statements results in `templating/parsing errors found`, which prevents it from fixing rule violations. Seems that this feature is not supported yet by Redshift dialect. 

### How to reproduce

Let's create a small script with a violation of `layout.spacing`, which I'd like to have fixed by SQLFluff. The script grants a scoped permission:
```bash
echo "GRANT    SELECT FOR TABLES IN DATABASE analytics TO developer_user;" > example.sql
```

Now it's time to fix it:
```bash
uv run sqlfluff fix ./example.sql
```

Which yields:
```
==== finding fixable violations ====
  [1 templating/parsing errors found]                                                              
  == [example.sql] ==
  L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'GRANT   
                       | SELECT FOR TABLES IN DATABASE a...'
==== no fixable linting violations found ====
All Finished 📜 🎉!
```

The statement in `example.sql` is valid, and was tested on a 2 x ra3.large Redshift cluster running patch 204.

### Dialect

Redshift

### Version

`python==3.11.9`
`sqlfluff[rs]==4.3.0`

### Configuration

pyproject.toml:
```toml
[tool.sqlfluff.core]
dialect = "redshift"
rules = ["layout.spacing"]
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
